### DISCUSSION TITLE
Avoiding cache while fetching on mount once

### DISCUSSION THREAD

**Discussion by cubecleveland** [TanStack/query#2171] [General]
HI Guys,

I am sure this question was asked before under other cirumcrtences...
I want to fire a query and NOT save it in cache at all......
So the query will fire every time on remount WITHOUT using the cache...

I looked at staletime but i think the query still stays in the chache...

Thanks,

**Comment by TkDodo** [comment]
if the query is _not_ in the cache, it would need to re-fetch on every _render_ - that wouldn't be good :/

there are ways to do it, but what's your use-case please?

**Reply by cubecleveland** [reply]
HI,

I have a query that fetches a data from a remote Physical Device.
We need the real time info that on the device since it changes rapidlly.
I prefer no to show the user any prefecthed data at all when he loads the page....

Hope that make sense...
R

**Reply by TkDodo** [reply]
so when a component mounts that has a useQuery, react-query will give the data from the cache and do a background refresh if the query is stale. since the default stale time is 0, there will be a background refresh out of the box whenever a component mounts. you'll see stale data instantly, and then it updates once the request comes back. does that cover your case?

**Reply by cubecleveland** [reply]
Kind of.... It will still render the data from cache..... and then will background refresh.. i want to avoid this altogether... literally bypassing the cache... is there any way to achive that ?

**Reply by TkDodo** [reply]
you can check for the `isFetching` flag, and if it's true, you render a loading spinner instead? It will also be true for background updates ...

if spinners are better than stale data and you don't want the cache for a reason, then I'd make this request without react-query. Never had the situation though....

**Reply by TkDodo** [reply]
cacheTime: 0 will also remove the data from the cache if all subscribers unmount - maybe that's enough?

**Reply by cubecleveland** [reply]
i think that worked.
I wish there was a cache:False option to completely avoid loggin the query altogether.

But good for now.
Thanks a million again!!!
